presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -306,15 +332,91 @@ TEST_F(UnsafeRowShuffleTest, endToEnd) {
     params.destination = i;
 
     bool noMoreSplits = false;
-    auto [taskCursor, serializedResults] = readCursor(params, [&](auto* task) {
+    auto [taskCursor, output] = readCursor(params, [&](auto* task) {
       if (noMoreSplits) {
         return;
       }
       addRemoteSplits(task, {leafTaskId});
       noMoreSplits = true;
     });
 
+    ASSERT_EQ(output.size(), 1);
+    auto vector= copyResultVector(output[0]);
+    outputVectors.push_back(vector);
+    ASSERT_FALSE(shuffle.hasNext(i)) << i;
+  }
+  velox::exec::test::assertEqualResults(inputVectors, outputVectors);
+}
+
+TEST_F(UnsafeRowShuffleTest, persistentShuffle) {
```

Comment:
What's the difference between this test and endToEnd? If it is only the implementation of shuffle interface, perhaps, both tests could share most of the code.

Reply:
These two and even fuzzer test are refactored now.